### PR TITLE
add eachsplit

### PR DESCRIPTION
**Pull request by longemen3000**
this adds https://github.com/JuliaLang/julia/pull/39245 to Compat.jl. i'm marking this PR as a draft because i don't have the correct version number yet 😅

### CODE REVIEW

File: src/Compat.jl [hunk 1]
```diff
@@ -1226,6 +1226,55 @@ if VERSION < v"1.8.0-DEV.300"
     end
 end
 
+# https://github.com/JuliaLang/julia/pull/39245
+if VERSION < v"1.8.0-DEV.300"
```
Comment (martinholters) [inline src/Compat.jl:1230]:
```suggestion
if VERSION < v"1.8.0-DEV.487"  
```

**Comment by longemen3000** [comment]
tests failed because the keyword syntax `f(x;kw1,kw2)` was introduced in 1.5  😅. now its ready

File: src/Compat.jl [hunk 2]
```diff
@@ -1237,6 +1237,56 @@ if VERSION < v"1.8.0-DEV.300"
     end
 end
 
+# https://github.com/JuliaLang/julia/pull/39245
+if VERSION < v"1.8.0-DEV.487"  
+    export eachsplit
+    function eachsplit end
```
Comment (martinholters) [inline src/Compat.jl:1243]: Doesn't hurt, but this declaration seems rather pointless.

File: test/runtests.jl [hunk 3]
```diff
@@ -1298,3 +1298,76 @@ end
...
+    @test eachsplit("a  b \t c\n"; keepempty=true) |> collect == ["a","","b","","","c",""]
+
+    let str = "a.:.ba..:..cba.:.:.dcba.:."
+    @test eachsplit(str, ".:.") |> collect == ["a","ba.",".cba",":.dcba",""]
+    @test eachsplit(str, ".:."; keepempty=false) |> collect == ["a","ba.",".cba",":.dcba"]
+    @test eachsplit(str, ".:.") |> collect == ["a","ba.",".cba",":.dcba",""]
+    @test eachsplit(str, r"\.(:\.)+") |> collect == ["a","ba.",".cba","dcba",""]
+    @test eachsplit(str, r"\.(:\.)+"; keepempty=false) |> collect == ["a","ba.",".cba","dcba"]
+    @test eachsplit(str, r"\.+:\.+") |> collect == ["a","ba","cba",":.dcba",""]
+    @test eachsplit(str, r"\.+:\.+"; keepempty=false) |> collect == ["a","ba","cba",":.dcba"]
```
Comment (martinholters) [inline test/runtests.jl:1347]: Indent these.

**Review by martinholters — APPROVED** [review]
LGTM modulo indentation of the tests.

File: src/Compat.jl (hunk 2 above)
Comment (longemen3000) [reply src/Compat.jl:1243]: oh, wait, it was for the docs

Comment (martinholters) [reply src/Compat.jl:1270]: Right. We mostly don't add docstrings in Compat, but we do have some, and I don't care much one way or the other.

**Pull request merged**